Form/Type/BasePickerType.php

```diff
@@ -31,11 +32,30 @@
     private $formatConverter;
 
     /**
+     * @var TranslatorInterface
+     */
+    protected $translator;
+
+    /**
+     * @var string
+     */
+    protected $locale;
+
+    /**
      * @param MomentFormatConverter $formatConverter
+     * @param TranslatorInterface $translator
      */
-    public function __construct(MomentFormatConverter $formatConverter)
+    public function __construct(MomentFormatConverter $formatConverter, TranslatorInterface $translator = null)
     {
         $this->formatConverter = $formatConverter;
+        $this->translator = $translator;
+
+        if ($this->translator !== null) {
+            $this->locale = $this->translator->getLocale();
+        } else {
+            @trigger_error('Initializing '.__CLASS__.' without TranslatorInterface is deprecated. The default locale used by BasePicker is now \'en\'', E_USER_DEPRECATED);
```

Comment:
"is deprecated since 2.4 and will be remove in 3.0."

> The default locale used by BasePicker is now \'en\''

I do not understand this sentence. I would prefer something like "Give TranslatorInterface on second parameter".